src/middlewared/middlewared/plugins/sysdataset.py

```diff
@@ -481,23 +497,17 @@ async def setup_datasets(self, pool, uuid):
             if is_cores_ds:
                 props['quota'] = '1G'
             if dataset not in datasets_prop:
-                await self.middleware.call('zfs.dataset.create', {
-                    'name': dataset,
-                    'properties': props,
-                })
-            elif is_cores_ds and datasets_prop[dataset]['used']['parsed'] >= 1024 ** 3:
+                await self.middleware.call('zfs.dataset.create', {'name': dataset, 'properties': props})
+            elif is_cores_ds and datasets_prop[dataset]['used']['value'] >= 1024 ** 3:
                 try:
                     await self.middleware.call('zfs.dataset.delete', dataset, {'force': True, 'recursive': True})
-                    await self.middleware.call('zfs.dataset.create', {
-                        'name': dataset,
-                        'properties': props,
-                    })
+                    await self.middleware.call('zfs.dataset.create', {'name': dataset, 'properties': props})
                 except Exception:
                     self.logger.warning("Failed to replace dataset [%s].", dataset, exc_info=True)
             else:
                 update_props_dict = {
                     k: {'value': v} for k, v in props.items()
-                    if datasets_prop[dataset][k]['value'] != v
+                    if datasets_prop[dataset][k]['raw'] != v
```

Comment:
Since you're **changing** value to raw, maybe add a comment to clarify why you're selecting raw over value.